src/util/coordinate_calculation.cpp

```diff
@@ -292,6 +292,33 @@ Coordinate interpolateLinear(double factor, const Coordinate from, const Coordin
     return {std::move(interpolated_lon), std::move(interpolated_lat)};
 }
 
+// compute the signed area of a triangle
+double signedArea(const Coordinate first_coordinate,
+                  const Coordinate second_coordinate,
+                  const Coordinate third_coordinate)
+{
+    return 0.5 * (-static_cast<double>(toFloating(second_coordinate.lon)) *
+                      static_cast<double>(toFloating(first_coordinate.lat)) +
+                  static_cast<double>(toFloating(third_coordinate.lon)) *
+                      static_cast<double>(toFloating(first_coordinate.lat)) +
+                  static_cast<double>(toFloating(first_coordinate.lon)) *
+                      static_cast<double>(toFloating(second_coordinate.lat)) -
+                  static_cast<double>(toFloating(third_coordinate.lon)) *
+                      static_cast<double>(toFloating(second_coordinate.lat)) -
+                  static_cast<double>(toFloating(first_coordinate.lon)) *
+                      static_cast<double>(toFloating(third_coordinate.lat)) +
+                  static_cast<double>(toFloating(second_coordinate.lon)) *
+                      static_cast<double>(toFloating(third_coordinate.lat)));
```

Comment:
Hm. Maybe make some tmp. vars for the calculation, this is somewhat hard to read.